pysat/_instrument.py

```diff
@@ -1053,32 +1053,34 @@ def _assign_funcs(self, by_name=False, inst_module=None):
             # Either flags are not specified, or this combo is not
             self._password_req = False
 
-    def __str__(self):
-
-        output_str = '\npysat Instrument object\n'
+    def __repr__(self):
+        # Print the basic Instrument properties
+        output_str = 'pysat Instrument object\n'
         output_str += '-----------------------\n'
         output_str += 'Platform: ' + self.platform + '\n'
         output_str += 'Name: ' + self.name + '\n'
         output_str += 'Tag: ' + self.tag + '\n'
         output_str += 'Satellite id: ' + self.sat_id + '\n'
```

Comment:
```suggestion
    def __repr__(self):
        # return a string representation that could be used to recreate object
        # https://stackoverflow.com/questions/1436703/difference-between-str-and-repr
        output_str = 'pysat.Instrument('
        output_str += '%r, %r' % (self.platform, self.name)
        output_str += ', sat_id=%r, tag=%r)' % (self.sat_id, self.tag)
```

Reply:
```
In [1]: import pysat                                                                                                                                                       

In [2]: test = pysat.Instrument('pysat', 'testing')                                                                                                                        

In [3]: test                                                                                                                                                               
Out[3]: pysat.Instrument('pysat', 'testing', sat_id='', tag='')

```

Reply:
This is lacking custom keywords and custom methods.

Reply:
Ok, I don't like adding the `kwargs` to `__repr__` because it contains more than just what the user specified.  However, including it is less harmful than excluding it.  Adding `custom` here is too much.  The user can easily get that info by typing `inst.custom`.  There's no short way of including all the possible ways someone has altered `inst` after the initial call.  So I've limited `__repr__` to only include things that can be specified during the initialization.